Cleanup pkg create template

sbreitbart-NOAA opened the pull request:
<!---
Thanks for opening a PR. This commented text will **NOT** appear in the final PR. Toggle between Write and Preview to see what your PR will look like without the comments.
-->

# What is the feature?
*

# How have you implemented the solution?
*

# Does the PR impact any other area of the project, maybe another repo?
*

Schiano-NOAA commented on tests/testthat/test-ID_tbl_width_class.R:
```diff
@@ -1,16 +1,9 @@
-  dat <- utils::read.csv(
-    file.path("std_model_output.csv")
+  dat <- convert_output(
+    output_file = file.path("fixtures", "ss3_models", "models", "Sablefish2015", "Report.sso"),
```
@sbreitbart-NOAA I actually just made this change to the test in the branch I'm working on but I did the exact same thing so I think it will be OK

Schiano-NOAA commented:
> ping @Schiano-NOAA I've checked off my assigned boxes in the create_template section of #260, so feel free to make your changes too. There are some more empty boxes, but I think we should address them towards the end

Perfect! I just finished up the automate-ss3-keywords branch and was just about to ping you for review so this is great timing. I'll also work on this branch to change create_template

Schiano-NOAA commented on R/create_template.R:
```diff
@@ -4,52 +4,50 @@
-#' will pull the last saved stock assessment report skeleton.
-#' Default is false.
 #' @param format Rendering format (pdf, html, or docx).
+#' @param type Type of report to build. Default is SAR.
```
@sbreitbart-NOAA Do you think I should add what SAR stands for in the documentation?

Schiano-NOAA commented:
@sbreitbart-NOAA I am finished with this branch and have ~hopefully~ fixed all of the bugs after all the changes. It won't pass CMD check most likely bc I have not adjusted the tests, but I think it would be easier to do in another branch with all other tests. Feel free to do a check and lmk when you are good with me merging into cleanup-pkg

sbreitbart-NOAA commented:
> Yes one I am no longer doing and another I am planning on adjusting all tests at once in a different branch. The last unchecked box in that grouping, I don't think I added.

👍 We added the last one a few weeks ago. I'll leave it unchecked and keep an eye on it to make sure it doesn't get removed when these branches are eventually merged into dev

sbreitbart-NOAA commented on R/create_tables_doc.R:
```diff
@@ -57,9 +55,14 @@ create_tables_doc <- function(subdir = getwd(),
+        # add_option = TRUE,
+        chunk_option = c(
+          "echo: false",
+          "warnings: false",
```
same here re: switching "warnings" for "warning"

sbreitbart-NOAA commented on R/create_template.R:
```diff
@@ -267,33 +241,68 @@ create_template <- function(
+  if (grepl("^pdf$|^html$", tolower(format))) {
+    format <- tolower(format)
+  } else if (grepl("docx", tolower(format))) {
+    cli::cl_warn("The docx format is not currently supported by asar. Defaulting to pdf")
```
```suggestion
    cli::cli_alert_warning("The docx format is not currently supported by asar. Defaulting to pdf")
```

sbreitbart-NOAA commented on R/create_template.R:
```diff
@@ -267,33 +241,68 @@ create_template <- function(
+      question1 <- readline("Did you mean -- docx?")
+      if (!interactive()) question1 <- "y"
+      if (regexpr(question1, "y", ignore.case = TRUE) == 1) {
+        cli::cl_warn("The docx format is not currently supported by asar. Defaulting to pdf")
```
```suggestion
        cli::cli_alert_warning("The docx format is not currently supported by asar. Defaulting to pdf")
```

sbreitbart-NOAA commented on R/create_citation.R:
```diff
@@ -14,11 +14,11 @@
 #' )
```
The example should have `\dontrun{}` around it or else it'll run. Since this example doesn't work, it gives an error

sbreitbart-NOAA commented on R/create_yaml.R:
```diff
@@ -6,9 +6,9 @@
+#' @param author_list A list of strings containing pre-formatted author names
 #' and affiliations that would be found in the format in a yaml of a quarto
-#' file when using cat(author_list).
+#' file when using cat(author_list[[i]]).
```
```suggestion
#' file when using `cat(author_list[[i]])`.
```

sbreitbart-NOAA commented on tests/testthat/_snaps/add_chunk.md:
```diff
@@ -5,20 +5,21 @@
-      #| warning: false 
-      #| eval: true 
+      #| echo:true 
+      #| warnings: false 
```
```suggestion
      #| warning: false 
```

sbreitbart-NOAA commented on tests/testthat/_snaps/add_chunk.md:
```diff
@@ -5,20 +5,21 @@
     Code
-      cat(add_chunk("plot(cars$speed, cars$distance)", echo = "true"))
+      cat(add_chunk("plot(cars$speed, cars$distance)", chunk_option = c("echo:true",
+        "warnings: false", "eval: true")))
```
```suggestion
        "warning: false", "eval: true")))
```

sbreitbart-NOAA reviewed (commented):
I've made some suggestions to fix things I noticed by looking at the code and the R cmd check warnings. Once those things are fixed, I think this can be merged!

Schiano-NOAA commented on R/create_template.R:
```diff
@@ -4,52 +4,57 @@
+#' @param author A character vector of author names with their accompanying
+#' affiliations. For example an Jane Doe at the NWFSC Seattle, Washington office
+#' would have an entry of c("Jane Doe"="NWFSC-SWA"). Information on NOAA offices 
+#' is found in a database located in the package. Keys to the office addresses 
```
Do you suggest I add the link?
